### ISSUE TITLE
[Azure Pipelines] `tools/wpt/ tests: Windows` broken for all PRs

### ISSUE THREAD

**Issue by stephenmcgruer** [web-platform-tests/wpt#25394]
This is the Python 2 version of `tools/wpt/ tests`, the Python 3.6 and 3.8 versions are not failing.

I've seen this fail across two different PRs, so I suspect its unrelated to those particular PRs.

Examples:
* https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=53880&view=logs&jobId=802ce073-f5e3-5667-fac8-56b586b322fa
* https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=53744&view=logs&jobId=802ce073-f5e3-5667-fac8-56b586b322fa&j=802ce073-f5e3-5667-fac8-56b586b322fa

Last lines of failure log:

```
2020-09-03T15:31:11.3528072Z __________________________ test_run_verify_unstable ___________________________
2020-09-03T15:31:11.3528828Z 
2020-09-03T15:31:11.3560301Z temp_test = <function make_test at 0x000000000E7F6C88>
2020-09-03T15:31:11.3560919Z 
2020-09-03T15:31:11.3561283Z     @pytest.mark.slow
2020-09-03T15:31:11.3561919Z     @pytest.mark.remote_network
2020-09-03T15:31:11.3562518Z     @pytest.mark.skipif(sys.version_info >= (3, 8) and sys.platform == 'darwin',
2020-09-03T15:31:11.3563304Z                         reason="multiprocessing test hangs in Python 3.8 on macOS (#24880)")
2020-09-03T15:31:11.3564050Z     def test_run_verify_unstable(temp_test):
2020-09-03T15:31:11.3564681Z         """Unstable tests should be reported with a non-zero exit status. Stable
2020-09-03T15:31:11.3565406Z         tests should be reported with a zero exit status."""
2020-09-03T15:31:11.3566124Z         if is_port_8000_in_use():
2020-09-03T15:31:11.3567046Z             pytest.skip("port 8000 already in use")
2020-09-03T15:31:11.3568033Z         unstable_test = temp_test("""
2020-09-03T15:31:11.3568756Z             test(function() {
2020-09-03T15:31:11.3570172Z                 if (localStorage.getItem('wpt-unstable-test-flag')) {
2020-09-03T15:31:11.3570853Z                   throw new Error();
2020-09-03T15:31:11.3571329Z                 }
2020-09-03T15:31:11.3571783Z     
2020-09-03T15:31:11.3572278Z                 localStorage.setItem('wpt-unstable-test-flag', 'x');
2020-09-03T15:31:11.3572902Z             }, 'my test');
2020-09-03T15:31:11.3573330Z         """)
2020-09-03T15:31:11.3573780Z     
2020-09-03T15:31:11.3574322Z         with pytest.raises(SystemExit) as excinfo:
2020-09-03T15:31:11.3575004Z             wpt.main(argv=["run", "--yes", "--verify", "--binary-arg", "headless",
2020-09-03T15:31:11.3575877Z                            "--channel", "dev", "chrome", unstable_test])
2020-09-03T15:31:11.3576531Z         assert excinfo.value.code != 0
2020-09-03T15:31:11.3576972Z     
2020-09-03T15:31:11.3577431Z         stable_test = temp_test("test(function() {}, 'my test');")
2020-09-03T15:31:11.3577952Z     
2020-09-03T15:31:11.3578590Z         with pytest.raises(SystemExit) as excinfo:
2020-09-03T15:31:11.3579329Z             wpt.main(argv=["run", "--yes", "--verify", "--binary-arg", "headless",
2020-09-03T15:31:11.3580008Z                            "--channel", "dev", "chrome", stable_test])
2020-09-03T15:31:11.3580671Z >       assert excinfo.value.code == 0
2020-09-03T15:31:11.3581166Z E       assert 1 == 0
2020-09-03T15:31:11.3581728Z E        +  where 1 = SystemExit(1,).code
2020-09-03T15:31:11.3582405Z E        +    where SystemExit(1,) = <ExceptionInfo SystemExit tblen=5>.value
2020-09-03T15:31:11.3583714Z 
2020-09-03T15:31:11.3584164Z tests\test_wpt.py:243: AssertionError
2020-09-03T15:31:11.3605567Z ---------------------------- Captured stdout call -----------------------------
2020-09-03T15:31:11.3606690Z ERROR Cannot enable MojoJS: Cannot determine the version of Chrome
2020-09-03T15:31:11.3607430Z ERROR Cannot enable MojoJS: Cannot determine the version of Chrome
2020-09-03T15:31:11.3629180Z ---------------------------- Captured stderr call -----------------------------
2020-09-03T15:31:11.3630376Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3630975Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3631640Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3632308Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3632943Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3633430Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3633984Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3634571Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3635259Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3636050Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3636662Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3637430Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3638050Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3638664Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3639273Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3640473Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3641113Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3641665Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3642254Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3642862Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3643403Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3644042Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3644650Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3645201Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3645928Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3646507Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3647119Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3647731Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3648427Z Get-Item : Cannot find path 'C:\Program Files (x86)\Google\Chrome\Application\chrome.exe' because it does not exist.
2020-09-03T15:31:11.3648922Z 
2020-09-03T15:31:11.3649213Z At line:1 char:2
2020-09-03T15:31:11.3649574Z 
2020-09-03T15:31:11.3649977Z + (Get-Item 'C:\Program Files (x86)\Google\Chrome\Application\chrome.ex ...
2020-09-03T15:31:11.3650386Z 
2020-09-03T15:31:11.3650808Z +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2020-09-03T15:31:11.3651231Z 
2020-09-03T15:31:11.3651752Z     + CategoryInfo          : ObjectNotFound: (C:\Program File...tion\chrome.exe:String) [Get-Item], ItemNotFoundExcep 
2020-09-03T15:31:11.3652283Z 
2020-09-03T15:31:11.3652535Z    tion
2020-09-03T15:31:11.3652876Z 
2020-09-03T15:31:11.3653298Z     + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand
2020-09-03T15:31:11.3653726Z 
2020-09-03T15:31:11.3653964Z  
2020-09-03T15:31:11.3654300Z 
2020-09-03T15:31:11.3654718Z Binary path C:\Program Files (x86)\Google\Chrome\Application\chrome.exe does not exist
2020-09-03T15:31:11.3655310Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3656089Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3656673Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3657423Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3657999Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3658556Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3659194Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3659739Z INFO:tools.wpt.utils:Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3660366Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3660948Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3661555Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3662160Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3696319Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3700370Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3701116Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3701752Z INFO:tools.wpt.utils:Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3702368Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3702979Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3703571Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3704165Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3704757Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3705791Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3706411Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3707032Z INFO:tools.wpt.utils:Set manifest_update to True
2020-09-03T15:31:11.3707763Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3708390Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3708928Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3709518Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3710056Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3710686Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3711287Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3711922Z INFO:tools.wpt.utils:Set manifest_download to True
2020-09-03T15:31:11.3712661Z Get-Item : Cannot find path 'C:\Program Files (x86)\Google\Chrome\Application\chrome.exe' because it does not exist.
2020-09-03T15:31:11.3713212Z 
2020-09-03T15:31:11.3713529Z At line:1 char:2
2020-09-03T15:31:11.3713894Z 
2020-09-03T15:31:11.3714327Z + (Get-Item 'C:\Program Files (x86)\Google\Chrome\Application\chrome.ex ...
2020-09-03T15:31:11.3714697Z 
2020-09-03T15:31:11.3715108Z +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2020-09-03T15:31:11.3715461Z 
2020-09-03T15:31:11.3716421Z     + CategoryInfo          : ObjectNotFound: (C:\Program File...tion\chrome.exe:String) [Get-Item], ItemNotFoundExcep 
2020-09-03T15:31:11.3716879Z 
2020-09-03T15:31:11.3717158Z    tion
2020-09-03T15:31:11.3717479Z 
2020-09-03T15:31:11.3717942Z     + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand
2020-09-03T15:31:11.3718318Z 
2020-09-03T15:31:11.3718582Z  
2020-09-03T15:31:11.3718877Z 
2020-09-03T15:31:11.3719326Z Binary path C:\Program Files (x86)\Google\Chrome\Application\chrome.exe does not exist
2020-09-03T15:31:11.3719958Z ------------------------------ Captured log call ------------------------------
2020-09-03T15:31:11.3720604Z INFO     tools.wpt.utils:utils.py:44 Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3721252Z INFO     tools.wpt.utils:utils.py:44 Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3721886Z INFO     tools.wpt.utils:utils.py:44 Set manifest_update to True
2020-09-03T15:31:11.3722533Z INFO     tools.wpt.utils:utils.py:44 Set manifest_download to True
2020-09-03T15:31:11.3723161Z INFO     tools.wpt.utils:utils.py:44 Set tests_root to D:\a\1\s
2020-09-03T15:31:11.3723828Z INFO     tools.wpt.utils:utils.py:44 Set metadata_root to D:\a\1\s
2020-09-03T15:31:11.3724590Z INFO     tools.wpt.utils:utils.py:44 Set manifest_update to True
2020-09-03T15:31:11.3725203Z INFO     tools.wpt.utils:utils.py:44 Set manifest_download to True
2020-09-03T15:31:11.3726086Z ------------- generated xml file: D:\a\1\s\tools\wpt\results.xml --------------
2020-09-03T15:31:11.3737009Z ========= 3 failed, 48 passed, 6 skipped, 1 xfailed in 58.94 seconds ==========
2020-09-03T15:31:11.5374880Z ERROR: InvocationError for command 'D:\a\1\s\tools\wpt\.tox\py27\Scripts\pytest.EXE' --junitxml=results.xml (exited with code 1)
2020-09-03T15:31:11.5383671Z ___________________________________ summary ___________________________________
2020-09-03T15:31:11.5384419Z ERROR:   py27: commands failed
2020-09-03T15:31:11.6303563Z ##[error]Cmd.exe exited with code '1'.
2020-09-03T15:31:11.6677308Z ##[section]Finishing: Run tests
```

**Comment by stephenmcgruer** [comment]
So looks like in https://github.com/web-platform-tests/wpt/blob/master/tools/wpt/browser.py#L651 we just assume that on Windows Chrome will be installed in `"$SYSTEMDRIVE\Program Files (x86)\Google\Chrome\Application\chrome.exe"`. On Azure, this is expanding (I assume) to `C:\Program Files (x86)\Google\Chrome\Application\chrome.exe`, and I assume that doesn't exist (either because the browser is no longer preinstalled, or because it's moved to be preinstalled somewhere else).

Indeed, `.azure-pipelines.yml` actually has this comment:

```
- job: wpt_integration_win
  displayName: 'tools/wpt/ tests: Windows'
  dependsOn: decision
  condition: dependencies.decision.outputs['test_jobs.wpt_integration']
  pool:
    vmImage: 'windows-2019'
  variables:
    HYPOTHESIS_PROFILE: ci
  steps:
  # full checkout required
  - task: UsePythonVersion@0
    inputs:
      versionSpec: '2.7.x'
  # currently just using the outdated Chrome/Firefox on the VM rather than
  # figuring out how to install Chrome Dev channel on Windows
  # - template: tools/ci/azure/install_chrome.yml
  # - template: tools/ci/azure/install_firefox.yml
  - template: tools/ci/azure/update_hosts.yml
  - template: tools/ci/azure/update_manifest.yml
  - template: tools/ci/azure/tox_pytest.yml
    parameters:
      directory: tools/wpt/
      toxenv: py27
```

So I'm assuming something changed and the 'outdated' Chrome/Firefox on the `windows-2019` vm is no longer there (or no long where we expect it to be).

cc @mustjab - this is really our (wpt's) problem, but do you happen to know if something changed recently with the azure vms such that there is no longer a preinstalled chrome, or if its changed locations?

**Comment by mustjab** [comment]
Based on this doc, Chrome should be installed on windows-2019 images: https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md and I'm not aware of any recent changes to this image, you can find all announcements here: https://github.com/actions/virtual-environments/labels/Announcement

If run a powershell task on the VM to search for chrome.exe, are you able to find any? Something like this should work: 

> pushd $SYSTEMDRIVE\
> cmd /c dir /s chrome.exe

**Comment by stephenmcgruer** [comment]
Looks like we're assuming it's in the x86 location, whereas from the PR @mustjab has awesomely figured out that it is at `C:\Program Files\Google\Chrome\Application\chrome.exe`. (Not `C:\Program Files (x86)\Google\Chrome\Application\chrome.exe`).

I'll send a workaround PR to check both locations, though ultimately I would really like to do something better than guess at file paths -_-.

**Comment by stephenmcgruer** [comment]
In fact, I just found the Chrome issue that was fixed and caused this to happen! https://crbug.com/380177#c63

**Comment by Hexcles** [comment]
Amazing find!

**Comment by stephenmcgruer** [comment]
Hexcles and I discussed, and we think the fix in #25467 is good enough to consider this issue closed.
